publish_courses/api.py

```diff
@@ -1,39 +1,60 @@
 # -*- coding: utf-8 -*-
 
 import logging
+from typing import Dict, List
 
+import boto3
+from botocore.exceptions import ClientError
+from canvas_sdk import RequestContext
+from canvas_sdk.exceptions import CanvasAPIError
+from canvas_sdk.methods.accounts import list_active_courses_in_account
+from canvas_sdk.utils import get_all_list_data
 from django.conf import settings
+from django.http import JsonResponse
+from lti_school_permissions.decorators import lti_permission_required_check
 from rest_framework import status
-from rest_framework.exceptions import (
-    PermissionDenied,
-    ValidationError as DRFValidationError)
-from rest_framework.generics import (
-    ListAPIView,
-    ListCreateAPIView)
+from rest_framework.exceptions import PermissionDenied
+from rest_framework.exceptions import ValidationError as DRFValidationError
+from rest_framework.generics import ListAPIView, ListCreateAPIView
 from rest_framework.permissions import BasePermission
 from rest_framework.response import Response
-from rest_framework.serializers import (
-    JSONField,
-    ModelSerializer)
+from ulid import ULID
 
-from async_operations.models import Process
 from bulk_utilities.bulk_course_settings import BulkCourseSettingsOperation
-from lti_school_permissions.decorators import lti_permission_required_check
-from publish_courses.async_operations import bulk_publish_canvas_sites
-from django.http import JsonResponse
-import simplejson as json
-from django.conf import settings
+from publish_courses.models import Job, JobDetails
 
 logger = logging.getLogger(__name__)
+
+SDK_SETTINGS = settings.CANVAS_SDK_SETTINGS
+# Make sure the session_inactivity_expiration_time_secs key isn't in the settings dict.
+SDK_SETTINGS.pop('session_inactivity_expiration_time_secs', None)
+SDK_CONTEXT = RequestContext(**SDK_SETTINGS)
+
 PC_PERMISSION = settings.PERMISSION_PUBLISH_COURSES
 
+# AWS configuration parameters from settings.
+AWS_REGION_NAME = settings.BULK_PUBLISH_COURSES_SETTINGS['aws_region_name']
+AWS_ACCESS_KEY_ID = settings.BULK_PUBLISH_COURSES_SETTINGS['aws_access_key_id']
+AWS_SECRET_ACCESS_KEY = settings.BULK_PUBLISH_COURSES_SETTINGS['aws_secret_access_key']
+QUEUE_NAME = settings.BULK_PUBLISH_COURSES_SETTINGS['job_queue_name']
+VISIBILITY_TIMEOUT = settings.BULK_PUBLISH_COURSES_SETTINGS['visibility_timeout']
 
-class ProcessSerializer(ModelSerializer):
-    details = JSONField()
+KW = {
```

Comment:
I might update this naming to something like `credentials` to make its purpose more clear.